tests/pipeline/test_pipeline_delegator.py

```diff
@@ -0,0 +1,58 @@
+import threading
+from typing import List, Tuple
+from unittest.mock import Mock
+
+from snuba.datasets.factory import get_dataset
+from snuba.datasets.plans.single_storage import SingleStorageQueryPlanBuilder
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage
+from snuba.pipeline.pipeline_delegator import PipelineDelegator
+from snuba.pipeline.simple_pipeline import SimplePipelineBuilder
+from snuba.query.parser import parse_query
+from snuba.request import Request
+from snuba.request.request_settings import HTTPRequestSettings
+from snuba.web import QueryResult
+
+
+def test() -> None:
+    cv = threading.Condition()
+    query_result = QueryResult({}, {"stats": {}, "sql": ""})
+    mock_query_runner = Mock(return_value=query_result)
+
+    def callback_func(args: List[Tuple[str, QueryResult]]) -> None:
+        with cv:
+            assert args == [("events", query_result), ("errors", query_result)]
+            cv.notify()
```

Comment:
I would make actually the callback `wait` so it is not executed until the main thread execute `notify`. Set a timeout for the `wait` call, which returns False if the timeout occurs. Then if wait returns false it means timeout and you fail the test.
The main thread receives the result, and only then notifies the condition, so we assert that we receive the results independently from the time it takes to the callback to run.